lib/irslackd.js

```diff
@@ -299,14 +310,98 @@ class Irslackd {
   }
   async onIrcPing(ircUser, msg) {
     // Send PONG
-    this.ircd.write(ircUser.socket, 'irslackd', 'PONG', [ 'irslackd' ]);
+    this.ircd.write(ircUser.socket, 'irslackd', 'PONG', ['irslackd']);
   }
   async onIrcMode(ircUser, msg) {
     let target = msg.args[0];
     if (target.substr(0, 1) === '#') {
-      this.ircd.write(ircUser.socket, 'irslackd', 'MODE', [ target ]);
+      this.ircd.write(ircUser.socket, 'irslackd', 'MODE', msg.args);
     }
   }
+  async onIrcInvite(ircUser, msg) {
+    const self = this;
+    try {
+      if (msg.args.length !== 2) {
+        throw Error('Must pass two params, the user to invite and the channel');
+      }
+      var userId = ircUser.ircToSlack.get(msg.args[0].substr(0, 1) === '@' ? msg.args[0].substr(1) : msg.args[0]);
+      var channelId = ircUser.ircToSlack.get(msg.args[1]);
+      if (!userId) throw Error('IRC nick not found in irslackd cache');
+      if (!channelId) throw Error('IRC channel not found in irslackd cache');
+      await ircUser.slackWeb.paginateCallOrThrow('conversations.invite', 'channel', {
+        users: userId,
+        channel: channelId,
+      });
+      self.ircd.write(ircUser.socket, 'irslackd', '341', [
+        ircUser.ircNick,
+        userId,
+        channelId,
+      ]);
+    } catch (e) {
+      var message;
+      if (this.isJson(e.message)) {
+        var errorFromData = typeof JSON.parse(e.message).error.data !== 'undefined' ? JSON.parse(e.message).error.data.error : 'no data element in error response';
+        message = ':/INVITE failed->' +
+        JSON.parse(e.message).error.code + '->' +
+        errorFromData;
+      } else {
+        message = ':/INVITE failed->' + e.message;
+      }
+      self.ircd.write(ircUser.socket, 'irslackd', '371', [
+        ircUser.ircNick,
+        message,
+      ]);
```

Comment:
I wonder if we can catch higher up in the stack and generalize the `371` response here? Would have to add some house-keeping to track the current IRC command but then we'd have it for everything instead of just `/invite`.

Reply:
i thought about that, but there are some places where i'm looping that i don't want to throw out to a more generalized catch, because i want to stay in loop.

i could make it a function i guess ... ?